controllers/topology/desired_state.go

```diff
@@ -170,6 +182,106 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeployment computes the desired state for a MachineDeploymentTopology.
+// The generated machineDeployment object is calculated using the values from the machineDeploymentTopology and
+// the machineDeployment class.
+func computeMachineDeployment(class *clusterTopologyClass, current *clusterTopologyState, topology clusterv1.MachineDeploymentTopology) (machineDeploymentTopologyState, error) {
```

Comment:
```suggestion
func computeMachineDeployment(class *clusterTopologyClass, current *clusterTopologyState, machineDeploymentTopology clusterv1.MachineDeploymentTopology) (machineDeploymentTopologyState, error) {
```

Reply:
eventually, you can use md instead of machineDeployments for all the local variables